Piwik.Tracker.Samples/PiwikTrackerSamples.cs

```diff
@@ -84,46 +83,46 @@ static private void RecordCustomVariables()
             piwikTracker.SetCustomVariable(1, "pagevar1", "pagevalue1", Scopes.Page);
             piwikTracker.SetCustomVariable(2, "pagevar2", "pagevalue2", Scopes.Page);
 
-            var response = piwikTracker.DoTrackPageView("Document title of current page view");
+            var response = await piwikTracker.TrackPageViewAsync("Document title of current page view");
 
             DisplayDebugInfo(response);
         }
 
         /// <summary>
         /// Records 2 custom dimensions
         /// </summary>
-        static private void RecordCustomDimensions()
+        private static async Task RecordCustomDimensionsAsync()
         {
-            var piwikTracker = new PiwikTracker(SiteId, PiwikBaseUrl);
+            var piwikTracker = new PiwikTracker(HttpClient, SiteId, PiwikBaseUrl);
             piwikTracker.SetUserAgent(UA);
 
             piwikTracker.SetCustomTrackingParameter("dimension1", "value1");
             piwikTracker.SetCustomTrackingParameter("dimension2", "value2");
 
-            var response = piwikTracker.DoTrackPageView("Document title of current page view");
+            var response = await piwikTracker.TrackPageViewAsync("Document title of current page view");
 
             DisplayDebugInfo(response);
         }
 
         /// <summary>
         ///  Records a simple page view with a specified document title
         /// </summary>
-        static private void RecordSimplePageView()
+        private static async Task RecordSimplePageViewAsync()

```

Comment:
I took the time to get more familiar with .NET async support.

Method calls in `private static void Main(string[] args)` have not been renamed, probably because they are commented.

When I uncomment and rename `RecordSimplePageView` to `RecordSimplePageViewAsync` in `Main()`, the compiler warns the keyword `await` should be used. As soon as the keyword `await` is added, the compiler warns the `Main()` method should be async, which is not possible. Is this expected?

Also, if I don't add the `await` keyword and I shutdown the Piwik server, no exception is reported. Is this expected? How one should handle exceptions?